### ISSUE TITLE
Site creation fails when using Japanese characters for site name

### ISSUE THREAD

**Issue by benlower** [tableau/server-client-python#50]
Trying to create a site with the following:
```python
    sites = "遊び場"
    new_site = TSC.SiteItem(name=site, content_url=site.replace(" ", ""), admin_mode=TSC.SiteItem.AdminMode.ContentAndUsers)
    server.sites.create(new_site)
```

This errors with:

	409002: Resource Conflict
	A site already exists with the content URL '遊び場'

The site doesn't exist on my server.

**Comment by RussTheAerialist** [comment]
@benlower Did you try this using postman directly?  That error is coming directly from the REST API, so I don't think this is a bug in the server client but in the REST api.  Can you verify via postman against the rest api directly, and update this issue once you have?

**Comment by t8y8** [comment]
@benlower specify a different content_url, that content_url is becoming "" (because you can't have Japanese characters as the content_url) and "" is the ID if the default site.

**Comment by t8y8** [comment]
Idea: we could validate that content_url only contains valid ascii characters and throws before making the call to Server

```
    @content_url.setter
    @property_not_nullable
    def content_url(self, value):
        if not is_ascii(value): raise ValueError
        self._content_url = value
```

And/or make it a new decorator

**Comment by RussTheAerialist** [comment]
I think making it a decorator would be better.  Something like "OnlyAscii" or something.

**Comment by t8y8** [comment]
@RussTheAerialist, I think according to SO this is the best py2/py3 way to do it:

```
def property_is_url_safe(func):
    @wraps(func)
    def wrapper(self, value):
        try:
            value.decode('ascii')
        except UnicodeDecodeError:
            raise ValueError("content_url must be ascii only and url safe")
        return func(self, value)

    return wrapper
```

**Comment by LGraber** [comment]
Why do you call is "is_url_safe"? That doesn't really describe what it is doing (I still have to go to the code to figure out that safe == ascii). Are you planning on adding other things here? It is also an odd property name given that it will be on a property which is not an url.  Besides that I am fine with it after talking with Russell

**Comment by RussTheAerialist** [comment]
Maybe have it be a property that takes an argument and call it something like "allowed_characters"?  And then define "allowed_content_url_characters" somewhere which includes a string (or list/tuple) of all of the allowed characters in a content url?

**Comment by t8y8** [comment]
Resolved in #64
